### PR TITLE
HS-582 Minion monitoring

### PR DESCRIPTION
**Pull request by yangli2014**
## Description

After the local environment startup, when querying the metrics with "response_time_msec" at http://localhost:19080, the minion instance metrics will be in the result list.  

## Jira link(s)
- https://issues.opennms.org/browse/HS-582

## Checklist
* [x] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [x] Appropriate reviewer(s) have been selected.
* [x] Jira issue(s) have been updated to "In Review".
* [x] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.

### CODE REVIEW

File: inventory/src/main/java/org/opennms/horizon/inventory/component/MinionRpcManager.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Component
+public class MinionRpcManager {
+    private static final String DEFAULT_TASK_RESULTS_TOPIC = "task-set.results";
+    private static final int DEFAULT_MESSAGE_SIZE = 1024;
+    private static final int MONITOR_INITIAL_DELAY = 3_000;
+    private static final int MONITOR_PERIOD = 30_000;
+    private static final long ECHO_TIMEOUT = 120_000;
```
Comment (cgorantla) [inline inventory/src/main/java/org/opennms/horizon/inventory/component/MinionRpcManager.java:71]: TIMEOUT could be 30_000

File: inventory/src/main/java/org/opennms/horizon/inventory/component/MinionRpcManager.java [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    private final MinionRpcClient rpcClient;
+    private final MonitoringSystemService service;
+    private final KafkaTemplate<String, byte[]> kafkaTemplate;
+
+    private int initialDelay = MONITOR_INITIAL_DELAY;
+    @Value("${kafka.topics.results:" + DEFAULT_TASK_RESULTS_TOPIC + "}")
+    private String kafkaTopic;
+    private boolean monitoringStarted = false;
```
Comment (cgorantla) [inline inventory/src/main/java/org/opennms/horizon/inventory/component/MinionRpcManager.java:79]: Is this only used for testing ? , if yes, we should avoid

File: inventory/src/main/java/org/opennms/horizon/inventory/model/LightMonitoringSystemDTO.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ *******************************************************************************/
+
+package org.opennms.horizon.inventory.model;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+@Getter
+public class LightMonitoringSystemDTO {
```
Comment (cgorantla) [inline inventory/src/main/java/org/opennms/horizon/inventory/model/LightMonitoringSystemDTO.java:36]: Can name it as `MonitoringSystemBean`

**Review by cgorantla — COMMENTED** [review]
lgtm! few comments

File: inventory/src/main/java/org/opennms/horizon/inventory/component/MinionRpcManager.java (hunk 2 above)
Comment (yangli2014) [reply inventory/src/main/java/org/opennms/horizon/inventory/component/MinionRpcManager.java:79]: removed

File: inventory/pom.xml [hunk 4]
```diff
@@ -213,6 +213,11 @@
             <version>${grpc.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>io.github.hakky54</groupId>
+            <artifactId>logcaptor</artifactId>
+            <version>2.7.10</version>
+        </dependency>
```
Comment (cgorantla) [inline inventory/pom.xml:220]: <scope>test</scope>

**Comment by yangli2014** [comment]
I will change the Monitor Type from Minion to ECHO.

**Review by cgorantla — APPROVED** [review]

**Pull request merged**